### ISSUE TITLE
distant incorrectly determines that hypens may not be used in hostnames

### ISSUE THREAD

**Issue by westernwontons** [chipsenkbeil/distant#197]
I have a distant server running on a remote machine, where the user is named 'ubuntu-user'. When I attempt to make the connection with the following command:

```bash
distant client launch ubuntu-user@123.456.78.99:22
```

I get the following error:

```bash
error: Invalid value "ubuntu-user@123.456.78.99:22" for '<DESTINATION>': Hostname label can only be a-zA-Z0-9 or hyphen ('-')

For more information try --help


Caused by:
    Hostname label can only be a-zA-Z0-9 or hyphen ('-')
```

**Comment by chipsenkbeil** [comment]
Which version of distant are you running?

**Comment by westernwontons** [comment]
Latest from master, I cloned the project and ran `cargo install --path .`

**Comment by westernwontons** [comment]
If you're open, I would be happy to contribute. I discovered this project yesterday and I really love the idea, plus it could be immensely useful for me because I have to manage a bunch of projects. It's always a pain to set everything up on a fresh system.

**Comment by chipsenkbeil** [comment]
@westernwontons yeah, I'm more than happy to have contributions! Feel free to take a look and submit a pull request if you figure it out.

You may want to wait for #200 to merge in. I'll be doing that as soon as the tests pass on the CI.

**Comment by chipsenkbeil** [comment]
The logic where validation is done should be here: https://github.com/chipsenkbeil/distant/blob/6ba3ded188fde098e3cac7edce212183dd29e99f/distant-net/src/common/destination/host.rs#L170

Specifically, you can see the error message here: https://github.com/chipsenkbeil/distant/blob/6ba3ded188fde098e3cac7edce212183dd29e99f/distant-net/src/common/destination/host.rs#L91

**Comment by chipsenkbeil** [comment]
Oh, and it could be that more than the host is being read into the destination parser, which can be found here: https://github.com/chipsenkbeil/distant/blob/6ba3ded188fde098e3cac7edce212183dd29e99f/distant-net/src/common/destination/parser.rs

**Comment by westernwontons** [comment]
After some tracking I've found that the problem seems to be here:

https://github.com/chipsenkbeil/distant/blob/4fb9045152ead6cbdee87f8cd67d9081ff765b0d/distant-net/src/common/destination/parser.rs#L72

The check when calling `is_alphanumeric` fails for hypens. I added a test with the username `some-user`. The `username` variable on line 72 only contained `some`,  because `parse_until` stops when the closure returns `false`.

If it's alright with you I could whip up a PR with some tests that would allow hypens
